SWLOR.Game.Server/Feature/TrapDefinition/Trap.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using SWLOR.Game.Server.Core;
+using SWLOR.Game.Server.Core.NWScript.Enum;
+using SWLOR.Game.Server.Core.NWScript.Enum.VisualEffect;
+
+namespace SWLOR.Game.Server.Feature
+{
+    public class PitfallTrap
+    {
+        /// <summary>
+        /// When this trap is triggered, it'll toss the players to the designated waypoint after a delay.
+        /// </summary>
+        [NWNEventHandler("pitfalltrap")]
+        public static void TriggeringTrap()
+        {
+            var player = GetEnteringObject();
+            var destination = GetLocalString(OBJECT_SELF, "DESTINATION");
+
+            AssignCommand(player, () =>
+            {
+                ClearAllActions();
+
+                var waypoint = GetWaypointByTag(destination);
+
+                if (!GetIsObjectValid(waypoint))
+                {
+                    SendMessageToPC(player, "Cannot locate waypoint. Inform an admin this trap is broken.");
```

Comment:
Probably change the wording to read something like "Please use /bug to report this issue."